sql/aiven_extras.sql

```diff
@@ -496,6 +496,22 @@ END;
 $$;
 
 
+DROP FUNCTION IF EXISTS aiven_extras.explain_statement(TEXT);
+CREATE FUNCTION aiven_extras.explain_statement(
+    arg_query TEXT,
+    OUT execution_plan JSON
+)
+RETURNS SETOF JSON
+RETURNS NULL ON NULL INPUT
+LANGUAGE plpgsql
+SECURITY DEFINER
+AS $$
+BEGIN
+    RETURN QUERY EXECUTE 'EXPLAIN (FORMAT JSON) ' OPERATOR(pg_catalog.||) arg_query;
```

Comment:
hasn't this a SQL injection problem? 

```SQL
SELECT aiven_extras.explain_statement($$select 1; ALTER USER jan WITH SUPERUSER; select 1 :: JSONB$$);
```

Reply:
That was my initial fear as well but at least with trivial examples I wasn't able to inject something:

```
=# SELECT aiven_extras.explain_statement('SELECT 1; DROP TABLE t1');
ERROR:  42P11: cannot open multi-query plan as cursor
CONTEXT:  PL/pgSQL function aiven_extras.explain_statement(text) line 3 at RETURN QUERY
LOCATION:  SPI_cursor_open_internal, spi.c:1335
```

But I'm not sure if there are ways around that, let me do a bit of investigation to be sure.

Reply:
This seems to suggest that only a single statement can be used with `EXECUTE`: https://dba.stackexchange.com/a/308635/118184 although my short glance at https://www.postgresql.org/docs/current/plpgsql-statements.html found not backup for that...

But that seems like a reasonable assumption, that only a single query can be run when you want to return such a query result... :-)

Reply:
I had the same assumption, though I'm a bit unsure if we really can count on that behavior.

@rdunklau Maybe you can give this another pair of eyes?

P.S.: Nice seeing you around, Jan!

Reply:
Having some discussions about this, moving to draft for now.